File: archive-application/src/main/java/site/archive/dto/v2/BannerResponseDto.java
package site.archive.dto.v2;

import site.archive.domain.banner.Banner;

public record BannerResponseDto(String type, String summaryImage, String mainContent) {

    public static BannerResponseDto from(Banner banner) {
        return new BannerResponseDto(banner.getType().toString(),
                                     banner.getSummaryImage(),
                                     banner.getMainContent());
    }

}

File: archive-application/src/main/kotlin/site/archive/dto/v2/BannerResponseDto.kt
package site.archive.dto.v2

import site.archive.domain.banner.Banner

data class BannerResponseDto(val type: String, val summaryImage: String, val mainContent: String) {

    companion object {
        @JvmStatic
        fun from(banner: Banner): BannerResponseDto {
            return BannerResponseDto(
                banner.type.toString(),
                banner.summaryImage,
                banner.mainContent
            )
        }
    }

}